awscli/alias.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import logging
+import os
+import shlex
+import subprocess
+
+from botocore.configloader import raw_config_parse
+
+from awscli.utils import emit_top_level_args_parsed_event
+
+
+LOG = logging.getLogger(__name__)
+
+
+class InvalidAliasException(Exception):
+    pass
+
+
+class AliasLoader(object):
+    ALIAS_FILENAME = os.path.expanduser(
+        os.path.join('~', '.aws', 'cli', 'alias'))
+
+    def __init__(self, alias_filename=None):
+        """Interface for loading and interacting with alias file
+
+        :param alias_filename: The name of the file to load aliases from.
+            This file must be an INI file. If no filename is provided,
+            the value of ALIAS_FILENAME will be used.
+        """
+        self._filename = alias_filename
+        if alias_filename is None:
+            self._filename = self.ALIAS_FILENAME
+        self._aliases = self._load_aliases()
+        self._cleanup_alias_values()
+
+    def _load_aliases(self):
+        if os.path.exists(self._filename):
+            return raw_config_parse(
+                self._filename, parse_subsections=False)
+        return {'toplevel': {}}
+
+    def _cleanup_alias_values(self):
+        aliases = self.get_aliases()
+        for alias in aliases:
+            # Beginning and end line separators should not be included
+            # in the internal representation of the alias value.
+            aliases[alias] = aliases[alias].strip(os.linesep)
```

Comment:
Do we have to preserve whitespace in addition to line separators?  Seems easier to just use `.strip()` without `os.linesep`.  It's still possible to have `\r` in files on posix.